src/text_files_splitter/text_files_splitter.py

```diff
@@ -0,0 +1,192 @@
+        size *= BytesMultiples.DECIMAL
+        metric -= 1
+
+    return size
+
+
+def get_total_file_splits(file_bytes, file_size_limit):
+
+    splits = file_bytes / file_size_limit
+    return int(splits) if splits.is_integer() else int(splits) + 1
+
+
+def delete_file_last_row(file):
+    file_rows_exceeded_size = open(file, 'r+').readlines()
+    file_rows_exceeded_size.pop()
+
+    file_finished = open(file, 'w+')
+    file_finished.writelines(file_rows_exceeded_size)
+
+
+def get_file_name_splitted(file_suffix, split_count):
+    return '{}_{}.csv'.format(file_suffix, split_count)
+
+
+def create_file_splitted(output_directory, file_splitted_name):
+    file_path = os.path.join(output_directory, file_splitted_name)
+    return open(file_path, 'w')
+
+
+def close_file(file_object):
+    file_object.close()
+
+
+class TextFilesSplitter():
+    def __init__(self, source_file, file_name, max_size_file, file_type='csv', metric_units=ListMetrics.B, output_directory='.', include_header=True, file_suffix=None):
+        self.source_file = source_file
+        self.file_name = file_name
+        self.max_size_file = max_size_file
+        self.file_type = file_type
+        self.metric_units = metric_units
+
+        if not os.path.isdir(output_directory):
+            os.mkdir(output_directory)
+
+        self.output_directory = output_directory
+        self.include_header = include_header
+        self.file_suffix = file_suffix
+
+    def append_new_file(self, file_size, count):
+        self.splitted_files.append({
+            "file_name": self.file_splitted_name,
+            "path": self.output_directory,
+            "file_size": file_size,
+            "count": count
+        })
+
+    def get_file_splitted(self):
+
+        self.splitted_files = []
+
+        file_path = os.path.join(self.source_file, self.file_name)
+
+        self._buffer = 1000000
+
+        if not os.path.exists(file_path):
+            raise FileNotFoundError(
+                'The file {} does not exist.'.format(file_path))
+
+        file_size_limit = get_size_to_metric(
+            self.max_size_file, self.metric_units[1])
+
+        if not self.file_suffix:
+            self.file_suffix = self.file_name
+
+        split_count = 1
+        file_to_split = open(file_path, 'r')
+
+        header_value = file_to_split.readline()
+
+        file_header = ("".join(header_value))
+
+        file_to_split.seek(0)
+
+        last_row = None
+        lines_to_append_file = []
+        current_splitted_file_size = 0
+        total_size = 0
+        line_count = 0
+
+        def file_splitted_setup(
+                current_splitted_file_size, total_size, first_file):
+            self.file_splitted_name = get_file_name_splitted(
+                self.file_suffix, split_count)
+            current_file_splitted = create_file_splitted(
+                self.output_directory, self.file_splitted_name)
+
+            if (self.include_header and not first_file):
```

Comment:
Booleans are code smells. 

In this case, if `include_header` is false the first file will contain the header but the other files wont' Which is wrong.

In your case is because when you are operating over the files you leave the header in the first one. It's simpler to extract the header and then continue working as usual:

```
header_value = file_to_split.readline()
file_header = ("".join(header_value))
file_to_split.seek(1)
```

Reply:
I mean, the implementation of the "include_header" functionality is wrong. If you want to add or not the header line in the output files... this behaviour is wrong if "include_header" is set to False. It doesn't work.

Add a test to check that when "include_header" is set to False the headers are not present in any of the splitted files and you will see what I mean.

Reply:
Thank you Gorka! I can see the error now!